hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -230,26 +226,19 @@ void tryStartJob(Function<Long, Long> executionIdSupplier) {
     private void rewriteDagWithSnapshotRestore(DAG dag, long snapshotId) {
         logger.info(jobIdString() + ": restoring state from snapshotId=" + snapshotId);
         for (Vertex vertex : dag) {
-            // items with keys of type BroadcastKey need to be broadcast to all processors
-            DistributedFunction<Entry<Object, Object>, ?> projection = (Entry<Object, Object> e) ->
-                    (e.getKey() instanceof BroadcastKey) ? new BroadcastEntry<>(e) : e;
             // We add the vertex even in case when the map is empty: this ensures, that
             // Processor.finishSnapshotRestore() method is always called on all vertices in
             // a job which is restored from a snapshot.
             String mapName = snapshotDataMapName(jobId, snapshotId, vertex.getName());
             Vertex readSnapshotVertex = dag.newVertex("__snapshot_read." + vertex.getName(), readMapP(mapName));
-            // We need a separate mapping vertex and can't use readMapP's projectionFn:
-            // the projection will cause key/value deserialization on partition thread, which doesn't have job's
-            // class loader. If the key/value uses a custom object, it will fail. For example, StreamKafkaP uses
-            // TopicPartition as the key or a custom processor can use custom key.
-            Vertex mapSnapshotVertex = dag.newVertex("__snapshot_map_." + vertex.getName(), mapP(projection));
+            Vertex explodeVertex = dag.newVertex("_snapshot_explode." + vertex.getName(), ExplodeSnapshotP::new);
```

Comment:
should be `__snapshot_` to match the name of the read vertex

Reply:
fixed